src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -343,13 +343,18 @@ public static String randomInvalidDomainName(int size)
     }
 
     public static String randomDomainName(int size)
+    {
+        return randomDomainName(size, null);
+    }
+
+    public static String randomDomainName(int size, @Nullable String prefix)
     {
         String domainName = "";
         do
         {
-            String prefix = randomString(1, null, ALPHANUMERIC_STRING); // domain needs to start with alphanumeric char
-            final String charset = ALPHANUMERIC_STRING + DOMAIN_SPECIAL_STRING;
-            domainName = prefix + randomString(size - 1, null, charset);
+            String _prefix = prefix != null ? prefix : randomString(1, null, ALPHANUMERIC_STRING); // domain needs to start with alphanumeric char
+            final String charset = prefix != null ? DOMAIN_SPECIAL_STRING : ALPHANUMERIC_STRING + DOMAIN_SPECIAL_STRING;
```

Comment:
Not sure why we'd want to skip the alphanumeric characters, but I'm sure there's a reason...

Reply:
That was just because I assumed that if a caller is providing their own prefix, they are using AlphaNumeric values and would just want to tack on the special chars. Let me know if you think I should put that back to always include `ALPHANUMERIC_STRING`.

Reply:
Seems simpler to not have the conditional there, but I don't feel strongly about it.